embeddings/hyperparameter_search/flair_configspace.py

```diff
@@ -20,8 +20,8 @@
 # Mypy currently properly don't handle dataclasses with abstract methods  https://github.com/python/mypy/issues/5374
 @dataclass  # type: ignore
 class AbstractFlairModelTrainerConfigSpace(BaseConfigSpace, ABC):
-    embedding_name: InitVar[Union[str, List[str]]]
-    param_embedding_name: Parameter = field(init=False)
+    model_name: InitVar[Union[str, List[str]]]
```

Comment:
I see that change as a step back. Because we use the `model` as part of the pipeline and not a part of the task or configspace. It's more confusing. I opt for `embedding_name`. I think it was already discussed. This comment refers not only to this file but in general.

Reply:
We don't limit the scope of our library only for finetuning. We can expect many models for other tasks. And therefore renaming is as `model_name` I consider as a not good idea in our case.

Reply:
Other option could be more descriptive name like `representation_model_name`

Reply:
Ok, I'll change it to embedding_name, the reason for picking model_name/model_name_or_path is that it is the name of the argument of transformers huggingface AutoModel

Reply:
Also keep in mind that static embedding config dataclasses have field model_name

Reply:
> Ok, I'll change it to embedding_name, the reason for picking model_name/model_name_or_path is that it is the name of the argument of transformers huggingface AutoModel

I understand your intuition. In the transformers library, their concept is clear, only one task model per transformer embedding. In our case, we should not limit to such approaches. And in our broader case, we can have multiple different task models for the same task. Therefore we need to differentiate between the text embedding model and the task model.

> Also keep in mind that static embedding config dataclasses have field model_name

Yes but in that case it's ok. Model name refers to Embedding. But in case of task it's not intuitive as it have multiple interpretation and could case misunderstanding.